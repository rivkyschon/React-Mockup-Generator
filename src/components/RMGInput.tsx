// src/components/RMGInput.tsx

import React from 'react';

interface RMGInputProps {
  placeholder?: string;
  type: string;
}

const RMGInput: React.FC<RMGInputProps> = ({ placeholder, type }) => {
  return (
    <input
      type={type}
      placeholder={placeholder}
      className="border-2 border-gray-300 p-2 rounded focus:outline-none focus:border-blue-500"
    />
  );
};

export default RMGInput;
