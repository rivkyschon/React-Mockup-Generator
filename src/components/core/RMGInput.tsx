// src/components/RMGInput.tsx

import React, {ChangeEventHandler} from 'react';

interface RMGInputProps {
  placeholder?: string;
  type?: string;
  value?: string;
  onChange?: (e: any) => void;
  onKeyDown?: (e: any) => void;
  className?: string;
}

const RMGInput: React.FC<RMGInputProps> = ({
  placeholder,
  type,
  className,
  value,
  onChange,
  onKeyDown,
}) => {
  return (
    <input
      type={type || 'text'}
      placeholder={placeholder || 'Type here..'}
      value={value}
      onKeyDown={onKeyDown}
      onChange={onChange}
      className={
        className ||
        'border-2 block border-gray-300 w-full p-3 rounded focus:outline-none focus:border-gray-500'
      }
    />
  );
};

export default RMGInput;
