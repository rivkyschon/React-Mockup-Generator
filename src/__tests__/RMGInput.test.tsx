// src/components/__tests__/RMGInput.test.tsx

import React from 'react';
import {render} from '@testing-library/react';
import RMGInput from '../components/core/RMGInput';

describe('RMGInput', () => {
  it('renders correctly', () => {
    const {asFragment} = render(
      <RMGInput type="text" placeholder="Enter text" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has the correct placeholder', () => {
    const {getByPlaceholderText} = render(
      <RMGInput type="text" placeholder="Enter text" />,
    );
    // expect(getByPlaceholderText('Enter text')).toBeInTheDocument();
  });
});
