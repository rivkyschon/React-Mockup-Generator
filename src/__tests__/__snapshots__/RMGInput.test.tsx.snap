// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RMGInput renders correctly 1`] = `
<DocumentFragment>
  <input
    class="border-2 block border-gray-300 w-full p-3 rounded focus:outline-none focus:border-gray-500"
    placeholder="Enter text"
    type="text"
  />
</DocumentFragment>
`;
